### DISCUSSION TITLE
How should I custom animation for choices

### DISCUSSION THREAD

**Discussion by HakkerBarry** [dialogic-godot/dialogic#2055] [Q&A] [answered]
Hi folks,

I'm trying to scale the textbox and make the portrait grey when the choices menu is showing up. After player make a decision, the textbox should scale back.  

Put those transition in a animation seem like the eaiest way. I want find a way to play the animation. It seems like the menu shows in the text event and the choice event start to execute after pick the choose. Is there any singal emit when the choice menu shows up and disappear?

**Comment by Jowan-Spooner** [comment, marked as answer]
Hey @HakkerBarry.

I assume you already are making a custom layout, if not, that is definitely needed for customization like this.

An animation player could certainly work for this. Tweens are another popular option for animating UI.

Here is a couple of useful signals to react to:

- Dialogic.Text.animation_textbox_show
- Dialogic.Text.animation_textbox_hide
- Dialogic.Choices.choices_shown (info:Dictionary)
- Dialogic.Choices.choice_selected (info:Dictionary)

These should be enough to activate all your animations I think.

Good luck!

**Reply by HakkerBarry** [reply]
Thank you so much!

**Reply by JoanfloresBassas** [reply]
Hi, I am currently customizing my Choice Layer and I was wondering if it's possible to personalize the hovered choice. Thanks
